migrations/2018-08-20-00-migrate-to-create-member-and-verify.ts

```diff
@@ -0,0 +1,293 @@
+ *
+ * All RequestInvite operations are transformed into CreateMember operations. This involves
+ * updating their op_code and updating the field called data.to_uid to data.torequest_invite_from_member_id.
+ *
+ * A Trust operation from the inviting member to the new invited member previously indicated that the
+ * new member was legitimate. Now, an explicit Verify operation is required. For every Trust operation
+ * between an inviting member and the person they invited, we create a new Verify operation. The Verify
+ * operation points to the new member's invite video as the Verification video, since all members created
+ * via the RequestInvite flow joined with joint invite videos where both the inviter and the invitee state
+ * their name.
+ *
+ * This migration is idempotent allowing it to be run again if new members go through the RequestInvite flow
+ * (as could continue to happen until the web-based flow is shut down).
+ *
+ * Note that this migration also modifies the GENESIS RequestInvite and Trust operations. These are the
+ * operations that led to the spontaneous creation and trusting of the GENESIS members (Mark and family).
+ *
+ * It adds the "GENESIS" special value to the request_invite_from_member_id field of the new GENESIS CreateMember
+ * operations - previously this field would have been null. Similarly, the Verify operation for these GENESIS
+ * members will have the creator_uid set to "GENESIS" instead of the previous value of null. The GENESIS
+ * Trust operations will be deleted, as they no longer have any purpose. THIS MEANS THAT THE REST OF THE
+ * CODEBASE MUST BE UPDATED WITH THE IDS OF THE NEW GENESIS VERIFY OPERATIONS AFTER THIS MIGRATION HAS BEEN RUN.
+ */
+
+import * as admin from "firebase-admin";
+import { Firestore } from "@google-cloud/firestore";
+
+import {
+  Operation,
+  OperationType,
+  VerifyOperation,
+  InviteOperation,
+  RequestVerificationOperation
+} from "@raha/api-shared/dist/models/Operation";
+import { Omit } from "@raha/api-shared/dist/helpers/Omit";
+
+import { getDb } from "./helpers";
+
+export const GENESIS_REQUEST_INVITE_OPS = [
+  "InuYAjMISl6operovXIR",
+  "SKI5CxMXWd4qjJm1zm1y",
+  "SUswrxogVQ6S0rH8O2h7",
+  "Y8FiyjOLs9O8AZNGzhwQ"
+];
+export const ORIGINAL_GENESIS_TRUST_OPS = [
+  "va9A8nQ4C4ZiAsJG2nLt",
+  "CmVDdktn3c3Uo5pP4rV6",
+  "uAFLhBjYtrpTXOZkJ6BD",
+  "y5EKzzihWm8RlDCcfv6d"
+];
+
+const GENESIS = "GENESIS";
+
+/**
+ * Generates an alphanumeric string token of length 10 or 11.
+ */
+export function generateToken() {
+  return Math.random()
+    .toString(36)
+    .slice(2)
+    .toString();
+}
+
+function getInviteVideoUrlForMemberId(memberId: string) {
+  return `https://storage.googleapis.com/raha-video/${memberId}/invite.mp4`;
+}
+
+async function migrateToCreateMemberAndVerify(db: Firestore, dryRun: boolean) {
+  const operationsCollection = db.collection("operations");
```

Comment:
Use the transaction to get the operations, to ensure it's consistent with the state of the db when the transaction occurs